File: app/src/main/java/com/arseniy/maps/activities/ListActivity.java
package com.arseniy.maps.activities;

import android.content.Intent;
import android.os.Bundle;


import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.ItemTouchHelper;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.arseniy.maps.data.CityAdapter;
import com.arseniy.maps.data.MapData;
import com.arseniy.maps.R;
import com.google.android.material.floatingactionbutton.FloatingActionButton;

import java.util.ArrayList;

public class ListActivity extends AppCompatActivity {
    private RecyclerView.Adapter mAdapter;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.city_list);
        RecyclerView recyclerView = findViewById(R.id.cityListView);

        recyclerView.setHasFixedSize(true);

        RecyclerView.LayoutManager layoutManager = new LinearLayoutManager(this);
        recyclerView.setLayoutManager(layoutManager);

        mAdapter = new CityAdapter(MapData.loadCityList());
        recyclerView.setAdapter(mAdapter);

        //FAB
        FloatingActionButton fab = findViewById(R.id.addCityFAB);
        fab.setOnClickListener(view -> startActivityForResult(new Intent(this, AddCityActivity.class), 0));


        ItemTouchHelper itemTouchHelper = new ItemTouchHelper(simpleItemTouchCallback);
        itemTouchHelper.attachToRecyclerView(recyclerView);
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        if (requestCode == resultCode) {
            final String cityName = data.getStringExtra("cityName");
            ArrayList<String> cityNamesList = ((CityAdapter)mAdapter).getData();
            cityNamesList.add(cityName);
            ((CityAdapter)mAdapter).setData(cityNamesList);
        }
    }

    private final ItemTouchHelper.SimpleCallback simpleItemTouchCallback = new ItemTouchHelper.SimpleCallback(0, ItemTouchHelper.LEFT | ItemTouchHelper.RIGHT) {
        @Override
        public boolean onMove(@NonNull RecyclerView recyclerView, @NonNull RecyclerView.ViewHolder viewHolder, @NonNull RecyclerView.ViewHolder target) {
            return false;
        }

        @Override
        public void onSwiped(@NonNull RecyclerView.ViewHolder viewHolder, int swipeDir) {
            final int cityId = viewHolder.getAdapterPosition();
            ArrayList<String> cityNamesList = ((CityAdapter)mAdapter).getData();
            MapData.removeCityFromStorage(cityNamesList.get(cityId)); //sharedPrefs
            cityNamesList.remove(cityId);
            ((CityAdapter)mAdapter).setData(cityNamesList); //list
        }
    };

}

File: app/src/main/java/com/arseniy/maps/activities/ListActivity.kt
package com.arseniy.maps.activities

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.ItemTouchHelper
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

import com.arseniy.maps.R
import com.arseniy.maps.data.CityAdapter
import com.arseniy.maps.data.MapData
import com.google.android.material.floatingactionbutton.FloatingActionButton

class ListActivity : AppCompatActivity() {
    private var mAdapter: RecyclerView.Adapter<*>? = null
    private val simpleItemTouchCallback = object : ItemTouchHelper.SimpleCallback(0, ItemTouchHelper.LEFT or ItemTouchHelper.RIGHT) {
        override fun onMove(recyclerView: RecyclerView, viewHolder: RecyclerView.ViewHolder, target: RecyclerView.ViewHolder): Boolean {
            return false
        }

        override fun onSwiped(viewHolder: RecyclerView.ViewHolder, swipeDir: Int) {
            val cityId = viewHolder.adapterPosition
            val cityNamesList = (mAdapter as CityAdapter).data
            MapData.removeCityFromStorage(cityNamesList!![cityId]) //sharedPrefs
            cityNamesList.removeAt(cityId)
            (mAdapter as CityAdapter).data = cityNamesList //list
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.city_list)
        val recyclerView = findViewById<RecyclerView>(R.id.cityListView)

        recyclerView.setHasFixedSize(true)

        val layoutManager = LinearLayoutManager(this)
        recyclerView.layoutManager = layoutManager

        mAdapter = CityAdapter(MapData.loadCityList())
        recyclerView.adapter = mAdapter

        //FAB
        val fab = findViewById<FloatingActionButton>(R.id.addCityFAB)
        fab.setOnClickListener { startActivityForResult(Intent(this, AddCityActivity::class.java), 0) }


        val itemTouchHelper = ItemTouchHelper(simpleItemTouchCallback)
        itemTouchHelper.attachToRecyclerView(recyclerView)
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, intent: Intent?) {
        super.onActivityResult(requestCode, resultCode, intent)
        if (requestCode == resultCode && intent != null) {
            val cityName = intent.getStringExtra("cityName")
            val cityNamesList = (mAdapter as CityAdapter).data
            cityNamesList!!.add(cityName)
            (mAdapter as CityAdapter).data = cityNamesList
        }
    }

}
